pac4j-kerberos/src/main/java/org/pac4j/kerberos/credentials/extractor/KerberosExtractor.java

```diff
@@ -0,0 +1,44 @@
+package org.pac4j.kerberos.credentials.extractor;
+
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.extractor.CredentialsExtractor;
+import org.pac4j.core.exception.CredentialsException;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.kerberos.credentials.KerberosCredentials;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * To extract Kerberos headers.
+ *
+ * @author Garry Boyce
+ * @since 2.1.0
+ */
+public class KerberosExtractor implements CredentialsExtractor<KerberosCredentials> {
+    private final String clientName;
+
+    public KerberosExtractor(final String clientName) {
+        this.clientName = clientName;
+    }
+
+    @Override
+    public KerberosCredentials extract(WebContext context) throws CredentialsException, HttpAction {
+        final String header = context.getRequestHeader(HttpConstants.AUTHORIZATION_HEADER);
+        if (header == null) {
+            throw HttpAction.unauthorizedNegotiate("Kerberos Header not found", context);
```

Comment:
yeah makes sense.
at least this time we gonna use an `IndirectClient`, so that's what I'll be focusing on.

> My only question is whether a direct client makes sense for Kerberos, but you know better...


good question for which I don't know a full answer. but if one is using Kerberos as a webservice/headless auth, the current `401` flow in direct client might make sense more than redirect with indirect client.
So to keep more flexibility for others, I'd just add an option in `DirectClient` -
 `FORCE_AUTH_IF_NO_CREDENTIALS` if true, when credentials are missing, a 401 should be returned.